packages/data-models/src/iri/iri-provider.js

```diff
@@ -0,0 +1,96 @@
+/** @module iriProvider */
+import Sha1 from '@dmodels/sha1.js'
+import { v4 as uuidv4 } from 'uuid'
+
+export default class IRIProvider {
+  /**
+   * Creates an Internationalized Resource Identifier (IRI) according to the options provided.
+   *
+   * @param identityData.identityData
+   * @param {object} [identityData] - an object containing information that will be used
+   * to generate content-based IRI, such as MD5 hash. identityData keys will be sorted in
+   * an ascending alphanumerical order and the values will be concatenated into a string.
+   * This string will then be used to compute an IRI. Same identityData objects
+   * will have the same IRI.
+   * @param {IRIProvider.IRITypes} [type=IRIProvider.IRITypes.AUTO] - A type of the IRI to create.
+   * Possible values:
+   * - IRIProvider.IRITypes.AUTO: if identityData is provided, will calculate the MD5 Hash IRI.
+   * Otherwise, will return a UUID version 4.
+   * @param identityData.type
+   * @returns {string} - A newly created IRI.
+   */
+  static getIRI ({ identityData = {}, type = IRIProvider.IRITypes.AUTO } = {}) {
+    if (type === IRIProvider.IRITypes.AUTO) {
+      if (IRIProvider._isValidIdentityData(identityData)) {
+        return IRIProvider._getMD5Hash(identityData)
+      } else {
+        return IRIProvider._getUUIDv4()
+      }
+    } else if (type === IRIProvider.IRITypes.MD5_HASH) {
+      if (!IRIProvider._isValidIdentityData(identityData)) {
+        throw new Error(IRIProvider.errMsgs.NO_IDENTITY_DATA)
+      }
+      return IRIProvider._getMD5Hash(identityData)
+    } else if (type === IRIProvider.IRITypes.UUID_V4) {
+      return IRIProvider._getUUIDv4()
+    }
```

Comment:
else throw error?

Reply:
It is only a suggestion :)
I don't like long if/else if/else constructions - they are not easy to read, and it is a "headache" to expand them.
May be it is better to create a simple const list like
```
 IRIType: { method: getIRIMethod, needCheckForIdentity: {Boolean} }
 ```
 
 We use such lists in different places in our code - and it is rather useful.
 @kirlat , what do you think?

Reply:
Definitely, we should do that. I've updated the code.